src/main/java/seedu/address/model/lesson/LessonWithoutOwner.java

```diff
@@ -0,0 +1,161 @@
+package seedu.address.model.lesson;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.DayOfWeek;
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.address.model.person.Person;
+
+public class LessonWithoutOwner implements Comparable<LessonWithoutOwner> {
```

Comment:
honestly right, I don't think `LessonWithoutOwner` is the way to go about this. It represents both a Recurring and Makeup lesson which defeats the abstraction purpose of us separating them into two different classes later on. Even if you add a separate withoutOwner class for each one, there's too much repetition and complicates the code too much.
If the only purpose of storing Person reference in Lesson is to create an entry with name, then this way is probably not ideal.

I thought of another way about this:
- `Lesson` does not store `Person` or `calendarEntry`. But we can have a `LessonEntry` object that stores the corresponding `Lesson` reference and take in `Person` during instantiation to get name. 
- So the calendar entries type is `Entry<LessonEntry>` and we store a list of entries in AddressBook. 
- To delete the entry corresponding to a lesson, we just look through the whole list for an entry with the lesson.

It'll mean a lot of changes so i also don't know what's best. What do you think?

Reply:
@lingshanng I didn't like the idea of having to store a list of entries/lessons because implementing a new parser for them is overly complicated, which was why I use a `LessonWithoutOwner` class to build a valid lesson, something like the `EditPersonDescriptor` that the `EditCommand` uses. I thought a lot about the various ways I could implement this feature and I think that this is the simplest way, plus our storage structure wouldn't need to change at all. The other simple alternative was to use `Optional` and allow `Lesson` to not have an owner/calendar entry, but this is not defensive as there is a possibility of creating and lesson without an owner (does not enforce compulsory associations).
I think of it as a `Lesson` builder class, which can build valid a `Lesson` (either recurring or makeup) which I've redefined to require an owner + calendar entry, like this:
                                                                     `MakeupLesson`
                                        builds                 /
`LessonWithoutOwner` ---------> `Lesson`        or
                                                                   \
                                                                    `RecurringLesson`
so I feel it doesn't cause too much of a problem for our abstraction(?) 
I agree there's a bit of repetition going due to having `isClashing` in both `LessonWithoutOwner` and `Lesson`. I think the easiest way to fix it is to modify the way `Lesson` subclasses inherit from it. I'm hesitant in redoing this entire feature and adding complexity for what feels like marginal benefits :/